contracts/FeralfileArtworkV4_2.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import { FeralfileExhibitionV4_1 } from "./FeralfileArtworkV4_1.sol";
+import { Base64 } from "@openzeppelin/contracts/utils/Base64.sol";
+
+contract FeralfileExhibitionV4_2 is FeralfileExhibitionV4_1 {
+
+    error TokenIDNotFound();
+    error MintNotEnabled();
+
+    struct TokenInfo {
+        string thumbnail;
+        bytes parameters;
+    }
+
+    struct MintDataWithIndex {
+        uint256 seriesId;
+        uint256 tokenId;
+        address owner;
+        uint256 tokenIndex;
+    }
+
+    mapping(uint256 => TokenInfo) public _tokenInfos; // tokenID => tokenInfo
+    mapping(uint256 => uint256) public _tokenIndexes; // tokenID => tokenIndex
+
+    string public tokenPlaceholderURL;
+    string public tokenPlaceholderThubmnail;
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        bool burnable_,
+        bool bridgeable_,
+        address signer_,
+        address vault_,
+        address costReceiver_,
+        string memory contractURI_,
+        uint256[] memory seriesIds_,
+        uint256[] memory seriesMaxSupplies_
+    )
+        FeralfileExhibitionV4_1(
+            name_,
+            symbol_,
+            burnable_,
+            bridgeable_,
+            signer_,
+            vault_,
+            costReceiver_,
+            contractURI_,
+            seriesIds_,
+            seriesMaxSupplies_
+        )
+    {}
+    
+    /// @notice Mint new collection of Artwork
+    /// @dev the function iterates over the array of MintDataWithIndex to call the internal function mintArtworksWithIndex
+    /// @param data an array of MintDataWithIndex
+    function mintArtworksWithIndex(
+        MintDataWithIndex[] calldata data
+    ) external onlyAuthorized {
+        if (!mintable) {
+            revert MintNotEnabled();
+        }
+
+        for (uint256 i = 0; i < data.length; i++) {
+            _tokenIndexes[data[i].tokenId] = data[i].tokenIndex;
+            _mintArtwork(data[i].seriesId, data[i].tokenId, data[i].owner);
+        }
+    }
+
+    /// @notice Update the thumbnail & parameters of an edition to a new value
+    function updateTokenInformation(uint256 tokenId, string calldata thumbnail, bytes calldata parameters)
+        external
+        onlyAuthorized
+    {
+        if (!_exists(tokenId)) {
+            revert TokenIDNotFound();
+        }
+
+        _tokenInfos[tokenId] = TokenInfo(thumbnail, parameters);
+    }
+
+    /// @notice _buildArtworkData returns an object of artwork which would push to the actually artwork
+    /// @param parameters - the parameters for building artwork data
+    function _buildArtworkData(bytes memory parameters)
```

Comment:
This function looks like redundant since it's really simple. Maybe just use `Base64.encode(parameters)` is OK